core/services/head_tracker.go

```diff
@@ -460,10 +460,10 @@ func (ht *HeadTracker) receiveHeaders(ctx context.Context) error {
 
 				err := ht.handleNewHead(ctx, blockHeader)
 				if ctx.Err() != nil {
-					return nil
+					return ctx.Err()
```

Comment:
This would mean that the parent context (which is derived from the ht.done channel) has been canceled which only happens on shutdown. We should add a comment to that effect

Reply:
ah yeah, i somehow thought it's a new context

Reply:
We want to return nil here, returning an error logs errors all the way down, but a context being cancelled is not an error.